gateway/src/main/java/io/zeebe/gateway/impl/broker/RoundRobinDispatchStrategy.java

```diff
@@ -25,10 +25,16 @@ public int determinePartition() {
     final BrokerClusterState topology = topologyManager.getTopology();
 
     if (topology != null) {
-      final int offset = partitions.getAndIncrement();
-      return topology.getPartition(offset);
-    } else {
-      return BrokerClusterState.PARTITION_ID_NULL;
+      // go over all partitions once
```

Comment:
Can be removed, or reworded to better explain why we're skipping non-leaders. Though maybe we should just comment the class, as you mentioned this is only sort of round-robin now.